packages/journey-manager/src/index.ts

```diff
@@ -458,6 +481,14 @@ export const run = async (props: RunProps): Promise<RunOutput> => {
         )
         .map((trigger) => ({ trigger, elements: getAll(trigger.query) }));
     }
+    if (eventType === "enterViewport") {
+      return enterViewportSteps
+        .filter(
+          ({ urlCondition }) =>
+            urlCondition == null || matchesUrlCondition(urlCondition),
+        )
+        .map((trigger) => ({ trigger, elements: getAll(trigger.query) }));
```

Comment:
I'd extract the conditional and processing steps here. So something like:

```
const stepsForEvent = (
    eventType: ActiveTriggerEventType,
  ) : StepWithQuery[] => {
   switch(eventType) {
     case "click": return clickSteps;
     case "enterViewport": return enterViewportSteps;
  }
}

const findActiveTriggers = (
    eventType: ActiveTriggerEventType,
  ): ActiveTrigger[] => 
       stepsForEvent(eventType)
       .filter(
          ({ urlCondition }) =>
            urlCondition == null || matchesUrlCondition(urlCondition),
        )
        .map((trigger) => ({ trigger, elements: getAll(trigger.query) }));
```

Reply:
Wound up with this solution:

```ts
  const findActiveTriggers = (
    eventType: ActiveTriggerEventType,
  ): ActiveTrigger[] => {
    const steps =
      eventType === "click"
        ? clickSteps
        : eventType === "enterViewport"
          ? enterViewportSteps
          : [];
    return steps
      .filter(
        ({ urlCondition }) =>
          urlCondition == null || matchesUrlCondition(urlCondition),
      )
      .map((trigger) => ({ trigger, elements: getAll(trigger.query) }));
  };

```